bigtable/tests/data_integration_test.cc

```diff
@@ -289,3 +289,79 @@ TEST_F(DataIntegrationTest, TableReadModifyWriteAppendValueTest) {
   CheckEqualUnordered(expected_cells_ignore_timestamp,
                       actual_cells_ignore_timestamp);
 }
+
+TEST_F(DataIntegrationTest, TableReadModifyWriteRowIncrementAmountTest) {
+  std::string const table_name = "table-read-modify-write-row-increment-test";
+  auto table = CreateTable(table_name, table_config);
+  std::string const key = "row-key";
+
+  // An initial; BigEndian int64 number with value 0.
+  std::string v1("\x00\x00\x00\x00\x00\x00\x00\x00", 8);
+  std::vector<bigtable::Cell> created{{key, family1, "c1", 0, v1, {}}};
```

Comment:
Can we have some function which can convert BigEndian int64 byte value in a proper numeric format.
May be some overloaded function in bigtable::Cell

Reply:
Good idea.  Unfortunately there is no way to tell whether the buffer is holding an BigEndian int64 value or simply a string with 8 characters, or an int64 integer but in LittleEndian format, or maybe a IEEE-754 double precision number (which could also be little endian or big endian as I recall).

What I have done is print as a string (the most common case), but if there are non-printable characters we print the buffer as hex characters, and if the buffer is exactly 8 bytes we also print it as an integer.  The output is like this:

```
row_key=row-key, family=family2, column=d2, timestamp=0, value=<........>(hex:\x00\x00\x00\xff\x00\x00\x0b\xb8)[uint64:1095216663480], labels={}
```

Reply:
Is it possible to add new class members to bigtable::Cell as num_value, int64_value and populate this by converting from value when user requested specific type as return value or throw exception or return null if value is not convertible.
just the thought!
Or we can add this as a task and discuss in detail for more accurate solution when time permits.